core/templates/dev/head/pages/exploration_editor/simple_editor_tab/SimpleEditorShimServiceSpec.js

```diff
@@ -0,0 +1,210 @@
+      $provide.value('explorationInitStateNameService', stubs.
+        explorationInitStateNameService);
+      $provide.value('explorationStatesService', stubs.explorationStatesService
+        );
+    });
+  });
+
+  beforeEach(inject(function($injector) {
+    srv = $injector.get('SimpleEditorShimService');
+  }));
+
+  beforeEach(function() {
+    spyOn(stubs.explorationTitleService, 'saveDisplayedValue');
+    spyOn(stubs.explorationStatesService, 'saveStateContent');
+    spyOn(stubs.explorationStatesService, 'saveInteractionId');
+    spyOn(stubs.explorationStatesService, 'saveInteractionCustomizationArgs');
+    spyOn(stubs.explorationStatesService, 'saveInteractionAnswerGroups');
+    spyOn(stubs.explorationStatesService, 'saveInteractionDefaultOutcome');
+    spyOn(stubs.explorationStatesService, 'addState');
+  });
+
+  it('should return title', function() {
+    var title = srv.getTitle();
+    expect(title).toEqual(titleMemento);
+  });
+
+  it('should return Introduction HTML', function() {
+    var html = srv.getIntroductionHtml();
+    expect(html).toEqual(states[currentStateName].content[0].value);
+  });
+
+  it('should return current State Name', function() {
+    var name = srv.getInitStateName();
+    expect(name).toEqual(currentStateName);
+  });
+
+  it('should return All States Name', function() {
+    var names = srv.getAllStateNames();
+    expect(names).toEqual(Object.keys(states));
+  });
+
+  it('should return the given state', function() {
+    var givenState = currentStateName;
+    var state = srv.getState(givenState);
+    expect(state).toEqual(states[givenState]);
+  });
+
+  it('should return Content HTML for given state', function() {
+    var givenState = currentStateName;
+    var html = srv.getContentHtml(givenState);
+    expect(html).toEqual(states[givenState].content[0].value);
+  });
+
+  it('should change the displayed title and should save it', function() {
+    var newTitle = 'NewTitle';
+    srv.saveTitle(newTitle);
+    expect(stubs.explorationTitleService.displayed).toEqual(newTitle);
+    expect(stubs.explorationTitleService.saveDisplayedValue)
+    .toHaveBeenCalled();
+  });
+
+  it('should save the introduction HTML', function() {
+    var newIntroductionHtml = 'This is the new Introduction HTML';
+    var stateNewContent = [{
+      type: 'text',
+      value: newIntroductionHtml
+    }];
+    srv.saveIntroductionHtml(newIntroductionHtml);
+    expect(stubs.explorationStatesService.saveStateContent)
+    .toHaveBeenCalledWith(currentStateName, stateNewContent);
+  });
+
+  it('should save the interaction ID', function() {
+    var newInteractionID = 'abc123';
+    var stateName = currentStateName;
+    srv.saveInteractionId(stateName, newInteractionID);
+    expect(stubs.explorationStatesService.saveInteractionId)
+    .toHaveBeenCalledWith(stateName, newInteractionID);
+  });
+
+  it('should save the Customization Args', function() {
+    var newCustomizationArgs = '-a -b -c';
```

Comment:
Please use a valid example of customization args; do some console logging to see what the format is.